shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -528,6 +522,69 @@ protected String getInitialRoute() {
     }
   }
 
+  /**
+   * The path to the bundle that contains this Flutter app's resources, e.g., Dart code snapshots.
+   * <p>
+   * When this {@code FlutterActivity} is run by Flutter tooling and a data String is included
+   * in the launching {@code Intent}, that data String is interpreted as an app bundle path.
+   * <p>
+   * By default, the app bundle path is obtained from {@link FlutterMain#findAppBundlePath(Context)}.
+   * <p>
+   * Subclasses may override this method to return a custom app bundle path.
+   */
+  @NonNull
+  public String getAppBundlePath() {
```

Comment:
I'm assuming this is just all copied code from elsewhere

Reply:
Should be. There should be little, if any, new code in this PR. Did you have any concerns?

Reply:
No, I just meant I didn't read the rest of this file carefully because I assume it was already somewhere else.